src/CodeGeneration/ApiGenerator/Views/Descriptors/Descriptor.cshtml

```diff
@@ -0,0 +1,85 @@
+@inherits TemplatePage<DescriptorPartialImplementation>
+@using System
+@using System.Linq
+@using System.Collections.Generic
+@using RazorLight
+@using ApiGenerator.Domain
+@{
+	DescriptorPartialImplementation d = Model;
+	var names = d.CsharpNames;
+	var type = names.GenericOrNonGenericDescriptorName;
+	var concreteInterface = names.GenericOrNonGenericInterfaceName;
+}
+	///<summary>descriptor for @names.MethodName <pre>@d.OfficialDocumentationLink</pre></summary>
+	public partial class @Raw(type) @(Raw(string.Format(" : RequestDescriptorBase<{0},{1}, {2}>, {2}", type,names.ParametersName, concreteInterface)))
+	{ 
+		internal override ApiUrls ApiUrls => @(Raw(names.RequestName)).Urls;
+@foreach (Constructor c in d.Constructors)
+{
+<text>		@(Raw(CodeGenerator.Constructor(c)))
+</text>
+}
+		// values part of the url path
+@foreach (UrlPart part in d.Parts)
+{
+<text>		@(Raw(part.ClrTypeName)) @(Raw(names.InterfaceName)).@(part.InterfaceName) => Self.RouteValues.Get@(Raw(string.Format("<{0}>",part.ClrTypeName)))("@(part.Name)");
+</text>
+}
+@foreach (FluentRouteSetter c in d.GetFluentRouteSetters())
+{
+<text>
+		@(Raw(c.XmlDoc))
+		@(Raw(c.Code))
+</text>
+}
+		// Request parameters
+@foreach (var param in d.Params)
+{
+	var original = param.QueryStringKey;
+	//skip parameters already part of the path portion of the url
+	if (d.Parts.Any(p=>p.Name == original))
+	{
+		continue;
+	}
+	//we prefer this parameter to be explictly implemented on the request body
+	if (param.RenderPartial && (d.HasBody))
+	{
+		continue;
+	}
+	var t = @param.TypeHighLevel;
+	var tSuffix = (t == "bool" || t == "bool?") ? " = true" : "";
+	var typed = !string.IsNullOrEmpty(names.GenericsDeclaredOnDescriptor);
+	var g = typed ? names.GenericsDeclaredOnDescriptor.Replace("<", "").Replace(">", "") : "T";
+	var desc = param.DescriptionHighLevel.ToList();
+	
+	await IncludeAsync("Shared/XmlDocs.cshtml", desc);
+	if(!string.IsNullOrWhiteSpace(param.Obsolete))
+	{
+<text>		[Obsolete("Scheduled to be removed in 7.0, @param.Obsolete")]
+</text>
+	}
+<text>
+		public @Raw(type) @(param.ClsName)(@param.DescriptorArgumentType @param.ClsArgumentName@tSuffix) => Qs("@original", @(param.ClsArgumentName));
+</text>
+	if (param.IsFieldsParam)
+	{
+<text>
+		///<summary>@param.Description</summary>
+		public @Raw(type) @param.ClsName@(Raw(typed ? "" : "<T>"))(params @Raw("Expression<Func<" + g + ", object>>[]") fields) @Raw(typed ? "" : "where " + g + " : class") => Qs("@original", fields?@Raw(".Select(e=>(Field)e)"));
+</text>
+	}
+	else if (param.IsFieldParam)
+	{
+<text>
+		///<summary>@param.Description</summary>
+		public @Raw(type) @param.ClsName@(Raw(typed ? "" : "<T>"))(@Raw("Expression<Func<"+ g +", object>>") field) @Raw(typed ? "" : "where " + g + " : class") => Qs("@original", (Field)field);
+</text>
+	}
+}
+	@if (names.DescriptorNotFoundInCodebase)
+	{<text>
+		//TODO THIS METHOD IS UNMAPPED!
```

Comment:
I've always wondered if we ought to introduce a deliberate compiler error here instead, to prompt us to do the right thing and fix it.... thoughts?

Reply:
++ :100: I want to generate these into `Unmapped.Requests|Response.<MethodName>.cs` where we generate stubs for `Request` and `Response` with obsoletes with compiler errors on them.

This makes adding a new API easier and makes sure we won't release a stub.